src/libstore/misc.cc

```diff
@@ -200,6 +200,40 @@ void Store::queryMissing(const std::vector<DerivedPath> & targets,
             auto drv = make_ref<Derivation>(derivationFromPath(bfd.drvPath));
             ParsedDerivation parsedDrv(StorePath(bfd.drvPath), *drv);
 
+            if (!knownOutputPaths && settings.useSubstitutes && parsedDrv.substitutesAllowed()) {
+                // Try to realise the static output hashes.
+
+                // TODO: I'm not sure this logic is quite right, how do we
+                // ensure all the needed paths are present?
+
+                // TODO: Should we also do this->queryRealisation?
+
+                // TODO: Originally, this was only done if the drv was a CA drv.
+                // But you could have a non-CA drv with a CA input. So it may be
+                // necessary to gate this on the CA experimental feature?
```

Comment:
By "originally", I meant here for my first attempt. I found that I also needed to do this for non-CA derivations with non-known output paths. Is this something which should be gated on the CA experimental feature?

Reply:
Yes, I think we can `experimentalFeatureSettings.require(Xp::CaDerivations);` here, as it shouldn't happen if the feature isn't here (non-Ca with not known output paths should only happen if they depend on a CA derivation)